### ISSUE TITLE
feat: WaveForm for complete Audiofile

### ISSUE THREAD

**Issue by synchronisator** [alnitak/flutter_soloud#75]
**Description**

Its currently not possible to create a waveform of a complete audiofile.
This is an example of Audacity:
![grafik](https://github.com/alnitak/flutter_soloud/assets/10772789/8d4cf7f0-a4f0-4d08-87a7-3c60eeafb799)

To create something like this, there has to be a method to get all values from the file. 
At the moment there is only a way to get small parts of the currently playing audiopart.

Thanks

**Comment by alnitak** [comment]
2 new methods to the plugin have been added:
- readSamplesFromFile()
- readSamplesFromMem()

```text
  /// Read [numSamplesNeeded] audio data from a file equally spaced in time.
  /// The returned Float32List is not guaranteed to be [numSamplesNeeded] long.
  /// Each value in the returned Float32List is in the range -1.0 to 1.0 and
  /// their values are the average of audio data from the previous
  /// index sample.
  ///
  /// [completeFileName] the complete path to the audio file.
  /// [numSamplesNeeded] is not guaranteed to be the same length as the returned
  /// Float32List. This could happen if the [endTime] is greater than the audio
  /// lenght.
  /// [startTime] in seconds. Defaults to 0.
  /// [endTime] in seconds. Defaults to -1. If -1, the audio will be read until
  /// the end of the file.
  @mustBeOverridden
  Float32List readSamplesFromFile(
    String completeFileName,
    int numSamplesNeeded, {
    double startTime = 0,
    double endTime = -1,
  });
```
There is also `readSamplesFromMem()` which accepts a `Uint8List buffer` instead of `String completeFileName`. This works on all platforms but is useful for the web because the browser sandbox doesn't let you read files.

This is the result compared to the one from Audacity:
![waveform](https://github.com/user-attachments/assets/9f608146-f163-4c19-bb51-89e65399dfb7)

A [web demo](https://marcobavagnoli.com/flutter_soloud_wave_data/).

This new feature is available on the `feat_waveform` branch.

**Comment by alnitak** [comment]
This feature is landed in v2.1.5!
